### PR TITLE
docs: add more redirects & remove unused cloud pages

### PR DESCRIPTION
**Pull request by vbarda**

### CODE REVIEW

File: docs/mkdocs.yml [hunk 1]
```diff
@@ -56,7 +56,11 @@ plugins:
   - autorefs
   - redirects:
       redirect_maps:
+        'cloud/index.md': 'concepts/index.md#langgraph-platform'
+        'cloud/how-tos/index.md': 'how-tos/index.md#langgraph-platform'
         'cloud/concepts/api.md': 'concepts/index.md#langgraph-platform'
+        'cloud/concepts/cloud.md': 'concepts/langgraph_cloud.md'
```
Comment (eyurtsev) [inline docs/mkdocs.yml:62]: does this not use the architecture picture?

File: docs/mkdocs.yml [hunk 2]
```diff
@@ -56,7 +56,11 @@ plugins:
   - autorefs
   - redirects:
       redirect_maps:
+        'cloud/index.md': 'concepts/index.md#langgraph-platform'
+        'cloud/how-tos/index.md': 'how-tos/index.md#langgraph-platform'
         'cloud/concepts/api.md': 'concepts/index.md#langgraph-platform'
```
Comment (eyurtsev) [inline docs/mkdocs.yml:61]: api should probably map to langgraph server instead? and not to the concepts page?

File: docs/mkdocs.yml (hunk 1 above)
Comment (vbarda) [reply docs/mkdocs.yml:62]: it does, good catch

File: docs/mkdocs.yml (hunk 2 above)
Comment (vbarda) [reply docs/mkdocs.yml:61]: yep

File: docs/mkdocs.yml (hunk 1 above)
Comment (vbarda) [reply docs/mkdocs.yml:62]: actually nevermind, i was correct, there was a separate image for this, this one was unused

**Pull request merged**